modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -244,7 +260,7 @@ public synchronized void start() {
                 busyLock
         );
 
-        sqlSchemaManager.registerListener(prepareSvc);
+        sqlSchemaManager.registerListener(queryCache::clear);
```

Comment:
With versioned schema, there is no need to clear cache. Plans for the old schema versions could be used by time-traveling queries.
Let's avoid this, or add a TODO to drop this line later.